lnp2p/listen.go

```diff
@@ -59,29 +60,26 @@ func acceptConnections(listener *lndc.Listener, port int, pm *PeerManager) {
 		}
 
 		// Add the peer data to the DB if we don't have it.
-		if pi == nil {
+		if len(pi.Addr) == 0 { // weird hack, but should work
```

Comment:
The reason I don't like having this again is because then the logic for "is the lnaddr defined?" be to check the length as a string, but if we end up making LN addrs more complicated than just a plain primitive then we have a lot of work to do to change that code later on.

Reply:
I think as mentioned, we won't (and shouldn't) make LnAddrs more complex than a string because a ln address is just a string :) We do have different data structures in place that can handle abstractions off LnAddr (plain `string` in this PR), so I don't think that'd be a problem.

Regarding the comment itself, I don't like the hack myself, but it works for what it intends to do, hence my choice. Open to alternatives as well.

Reply:
This is hard because Go's type system is just so limited it's hard to encode semantics into the type system in a nice way.